pkg/aws/ec2/ec2.go

```diff
@@ -4,15 +4,86 @@ package ec2
 import (
 	"context"
 	"errors"
+	"fmt"
 	"time"
 
 	"github.com/aws/aws-k8s-tester/pkg/ctxutil"
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/autoscaling/autoscalingiface"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
 	"go.uber.org/zap"
 )
 
+// PollASGUntilRunning describes all EC2 instances for the specified ASG.
+// It waits until all instances are 'running'.
+func PollASGUntilRunning(
```

Comment:
Nice shared code! Totally optional, but I might reorient this code to be an object instead of a helper function. 
ec2/asgclient.go
```
asgClient := ec2.ASGClient{ ... }
asgClient.WaitUntilRunning()
```

Reply:
Yeah we need clean up aws polling functions. I left it as TODO for now. Thanks.